pkg/skaffold/deploy/helm.go

```diff
@@ -607,12 +676,10 @@ func pairParamsToArtifacts(builds []build.Artifact, params map[string]string) (m
 
 	for param, imageName := range params {
 		b, ok := imageToBuildResult[imageName]
-		if !ok {
-			return nil, fmt.Errorf("no build present for %s", imageName)
+		if ok {
+			paramToBuildResult[param] = b
```

Comment:
was this error not valid? it seems like we do still want to make sure all build artifacts are present.

Reply:
So this was confusing to me so I will do my best to explain it.

They way this function was written before meant that you could only have images name in your `values:` map.
Example:
```yaml
deploy:
  helm:
    releases:
    - name: myrelease
      chartPath: k8s/client
      values:
        image: "my/image/name/built/&/pushed/by/skaffold
        image2. "second/image"
```

If you  were to add a different value just to specify maybe a replica count this function would throw an error saying not all build artifacts would have been used. I think that should not be the behaviour and instead let you add more values if you desire but detect when a value corresponds to and artifact and only then replace it by the full value including the tag.

It is true that with this change you wouldn't be able to make sure all build artifacts are used but I don't think that should be a problem. I believe in the kustomize render something similar happens in that if an artifact is not being used it is simply ignored,

I may be wrong in the things stated here. If you please correct me and I will find a different way to inject the full image names (including tags) into the render function.

Reply:
ahh ok I see what's going on here. this is an issue that people have complained about before, and one I certainly think we should address: https://github.com/GoogleContainerTools/skaffold/issues/3316.

however, I'd like to do this in a separate PR because I think we need more tests for this, and since this is addressing a separate issue than this PR is meant to. I think what you have here is a good place to start, since it would allow the deploy to at least go through when values are provided that are not images. would you mind taking this out of this PR, and submitting it in a follow up so we can track it separately? 🙏

Reply:
Right on the spot!

This was exactly what I was trying to fix. I will move outside of this MR and try to tackle it separately.

Thanks for bringing it up